spec/models/accounting/quickbooks/data_extractor_spec.rb

```diff
@@ -158,5 +158,334 @@ def expect_line_item_amounts(amounts)
+                  'department_ref' => nil } },
+              { 'id' => '7',
+                'description' => 'Eba',
+                'amount' => '10.99',
+                'detail_type' => 'JournalEntryLineDetail',
+                'journal_entry_line_detail' => {
+                  'posting_type' => 'Credit',
+                  'entity' => {
+                    'type' => 'Customer',
+                    'entity_ref' => { 'value' => '1', 'name' => "Amy's Bird Sanctuary", 'type' => nil } },
+                  'account_ref' => { 'value' => prin_acct.qb_id, 'name' => prin_acct.name, 'type' => nil },
+                  'class_ref' => { 'value' => '5000000000000026437', 'name' => loan.id, 'type' => nil },
+                  'department_ref' => nil } },
+              { 'id' => '8',
+                'description' => 'Eba',
+                'amount' => '10.99',
+                'detail_type' => 'JournalEntryLineDetail',
+                'journal_entry_line_detail' => {
+                  'posting_type' => 'Credit',
+                  'entity' => {
+                    'type' => 'Customer',
+                    'entity_ref' => { 'value' => '1', 'name' => "Amy's Bird Sanctuary", 'type' => nil } },
+                  'account_ref' => { 'value' => random_acct.name, 'name' => random_acct.name, 'type' => nil },
+                  'class_ref' => { 'value' => '5000000000000026437', 'name' => loan.id, 'type' => nil },
+                  'department_ref' => nil } }],
+            'id' => '167',
+            'sync_token' => 0,
+            'meta_data' => {
+              'create_time' => '2017-04-18T10:14:30.000-07:00',
+              'last_updated_time' => '2017-04-18T10:14:30.000-07:00' },
+            'txn_date' => '2017-04-18',
+            'total' => '12.30',
+            'private_note' => 'Random stuff' }
+        end
+
+        it do
+          expect(txn.loan_transaction_type_value).to eq('other')
+          expect(txn.managed).to be false
+        end
+
+        describe 'interest with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'disbursement with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'repayment with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+
+        describe 'random with multiple debits and credits' do
+          it do
+            expect(txn.loan_transaction_type_value).to eq('other')
+            expect(txn.managed).to be false
+          end
+        end
+      end
+
+      describe 'too few line items' do
```

Comment:
This is actually an impossible scenario, since QB won't allow a txn to be saved if its credits and debits are not equal. So it's not possible to have a journal entry with only one line item. I would add another line item that debits a random account.